### ISSUE TITLE
[Bug] `Transport(Custom(MissingKey("gas_price")))` in TransactionRequest

### ISSUE THREAD

**Issue by anna-carroll** [alloy-rs/alloy#464]
### Component

transports

### What version of Alloy are you on?

0b889d9f

### Operating System

macOS (Apple Silicon)

### Describe the bug

I setup a RpcClient, EthereumSigner, and Provider as follows:
```
        let url: Url = Url::parse(HOLESKY.rpc_url)?;
        let wallet = load_aws_signer("KEY_ID", Some(HOLESKY.chain_id)).await?;

        let provider = ProviderBuilder::new()
            .layer(alloy_provider::layers::NonceManagerLayer)
            .signer(EthereumSigner::from(wallet))
            .provider(RootProvider::new(RpcClient::new_http(url.clone())));
```
then, I construct a TransactionRequest with a blob Sidecar attached, and attempt to send the transaction: 
```
        let data =
            MyContract::submitCall::new(()).abi_encode();

        let tx: TransactionRequest = TransactionRequest {
            to: Some(HOLESKY.my_contract),
            from: Some(address!(HOLESKY.signer)),
            input: TransactionInput::new(data.into()),
            transaction_type: Some(Uint::from(0x03)),
            max_fee_per_blob_gas: Some(Uint::from(10)),
            max_fee_per_gas: Some(Uint::from(10)),
            max_priority_fee_per_gas: Some(Uint::from(2)),
            gas: Some(Uint::from(24000)),
            sidecar: Some(sidecar),
            ..Default::default()
        };

        let result = provider.send_transaction(tx).await?;
```
Running this program yields the following error: 
```
Error: Transport(Custom(MissingKey("gas_price")))
```

**Comment by prestwich** [comment]
This _should_ be closed by #426 . I'll add a test for it

**Comment by prestwich** [comment]
```
    #[test]
    fn builds_non_legacy() {
        let sidecar = Default::default();

        let tx: TransactionRequest = TransactionRequest {
            to: Some(Address::ZERO),
            from: Some(Address::ZERO),
            input: bytes!("").into(),
            transaction_type: Some(0x03),
            max_fee_per_blob_gas: Some(10),
            max_fee_per_gas: Some(10),
            max_priority_fee_per_gas: Some(2),
            gas: Some(24000),
            sidecar: Some(sidecar),
            nonce: Some(0),
            ..Default::default()
        };

        dbg!(tx.build_unsigned().unwrap());
    }
```

minimal test. fails in main

**Comment by prestwich** [comment]
fails in #426. guess this will be fixed in #431
